### PR TITLE
Fix RocksDB build on iOS

### PR DESCRIPTION
**Pull request by rajivshah3**
# Description of change
Fixes RocksDB build on iOS by using our fork, which includes iotaledger/rust-rocksdb@acbf6b8. This applies the correct flags so RocksDB compiles successfully.

Unfortunately we will have to maintain a fork again. When we update RocksDB to 7.1.2 or later, we can use the upstream package (pending the merge of https://github.com/rust-rocksdb/rust-rocksdb/pull/637)

\cc @pascalbros

## Links to any relevant issues
N/A

## Type of change

- Bug fix (a non-breaking change which fixes an issue)
- Chore

## How the change has been tested
Builds successfully on `aarch64-apple-ios`

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -31,7 +31,9 @@ tokio = { version = "1.12.0", default-features = false }
 zeroize = { version = "1.2.0", default-features = false, features = ["zeroize_derive"] }
 
 # storage
-rocksdb = { git = "https://github.com/rust-rocksdb/rust-rocksdb", rev = "86d983987e7cafce90ad8a147b0b325e6007eba6", default-features = false, features = ["lz4"], optional = true }
+# Can use upstream again once iOS support is merged
```
Comment (Thoralf-M) [inline Cargo.toml]: Maybe add https://github.com/rust-rocksdb/rust-rocksdb/pull/637 here? Otherwise one might not know what this is about

**Review by Thoralf-M — DISMISSED** [review]

**Review by samuel-rufi — DISMISSED** [review]

**Comment by Thoralf-M** [comment]
@rajivshah3 can you resolve the conflict?

**Comment by thibault-martinez** [comment]
@rajivshah3 Your PR on the rocksdb repository is still draft, are you still working on it ?

**Comment by rajivshah3** [comment]
@thibault-martinez yep, I have to make sure this works on the latest version of RocksDB. The upstream repository has already updated to 7.x which may include breaking changes

**Review by Thoralf-M — APPROVED** [review]

**Comment by thibault-martinez** [comment]
@rajivshah3 This has now been merged in rocksdb, can you update the revision and the lock files please ?

**Comment by thibault-martinez** [comment]
Closing as https://crates.io/crates/rocksdb/0.19.0 has been released and contain this change.

**Pull request closed without merging**